### ISSUE TITLE
"Copy Reference" menu option

### ISSUE THREAD

**Issue by dstillman** [zotero/zotero-connectors#205]
https://forums.zotero.org/discussion/comment/274291/#Comment_274291

Translate, convert to CSL-JSON, run through a bundled citeproc-js in the background page, copy to clipboard. Style option in prefs, I guess, similar to Quick Copy in the client.

**Comment by bwiernik** [comment]
It might be nice if the connector could listen to the Quick Copy preference in the client so that the connector style preference changes if client Quick copy is changed.

**Comment by dstillman** [comment]
It could, but bundling citeproc-js lets us do it without the client, and it'd a little awkward to have the connector have one setting and then have that setting changed automatically just because Zotero was opened.

**Comment by dstillman** [comment]
Though, granted, this also gets into installed styles, so maybe doing it without the client is a bad idea anyway.

**Comment by bwiernik** [comment]
I imagine most users will have the client installed. For me, having one default style setting that is shared across the connector and client seems more intuitive and potentially less confusing than having two isolated settings.

If the client isn't installed, then the preference change question is moot. The connector would need its own directory of installed styles (or it might be more straightforward to have a limited number of locally-available styles and access others using the API?)

**Comment by dstillman** [comment]
Yeah, I think doing this without the client is just unnecessary. So no bundled citeproc-js, and we just send the translated JSON to the client and get back a reference with the Quick Copy pref (which can even use the existing site-specific settings).

**Comment by dstillman** [comment]
https://forums.zotero.org/discussion/102005/ux-suggestion-quick-copy-from-connector-message-interface
